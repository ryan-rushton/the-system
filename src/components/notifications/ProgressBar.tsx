import { motion } from 'framer-motion';
import styles from './ProgressBar.module.scss';

/**
 * A animated bar to indicate that the notification will close itself after the nominated duration.
 */
export function ProgressBar({
  duration,
  onCompletion,
}: {
  /** The time in seconds at which the animation will finish and onCompletion will be invoked. */
  duration?: number;
  /** A function that gets called after the duration has completed. */
  onCompletion?(): void;
}) {
  if (duration === undefined) {
    return null;
  }

  return (
    <div data-testid="progress-bar" className={styles.container}>
      <motion.div
        className={styles.progress}
        initial={{ width: '0%' }}
        animate={{ width: '100%', transition: { duration } }}
        onAnimationComplete={onCompletion}
      />
    </div>
  );
}
