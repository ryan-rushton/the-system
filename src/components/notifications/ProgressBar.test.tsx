import { render, screen } from '@testing-library/react';
import { describe, expect, test } from 'vitest';
import { ProgressBar } from './ProgressBar';

describe('NavMenu', () => {
  test('it renders', () => {
    render(<ProgressBar duration={0} />);

    expect(screen.getByTestId('progress-bar')).toBeInTheDocument();
  });

  test('it matches snapshot', () => {
    render(<ProgressBar duration={0} />);

    expect(screen.getByTestId('progress-bar')).toMatchSnapshot();
  });

  test('it renders nothing when no duration', () => {
    render(<ProgressBar />);

    expect(screen.queryByTestId('progress-bar')).not.toBeInTheDocument();
  });
});
