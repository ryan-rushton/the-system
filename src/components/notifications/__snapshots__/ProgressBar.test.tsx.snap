// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NavMenu > it matches snapshot 1`] = `
<div
  class="_container_c5e3a7"
  data-testid="progress-bar"
>
  <div
    class="_progress_c5e3a7"
    style="width: 0%;"
  />
</div>
`;
